core/src/replay_stage.rs

```diff
@@ -242,33 +247,30 @@ impl ReplayStage {
                 next_leader,
                 poh_slot
             );
-            // TODO: remove me?
-            cluster_info.write().unwrap().set_leader(&next_leader);
-
-            if next_leader == *my_pubkey && reached_leader_tick {
-                trace!("{} starting tpu for slot {}", my_pubkey, poh_slot);
-                datapoint_warn!(
-                    "replay_stage-new_leader",
-                    ("count", poh_slot, i64),
-                    ("grace", grace_ticks, i64)
-                );
 
-                let tpu_bank = Bank::new_from_parent(&parent, my_pubkey, poh_slot);
-                bank_forks.write().unwrap().insert(tpu_bank);
-                if let Some(tpu_bank) = bank_forks.read().unwrap().get(poh_slot).cloned() {
-                    assert_eq!(
-                        bank_forks.read().unwrap().working_bank().slot(),
-                        tpu_bank.slot()
-                    );
-                    debug!(
-                        "poh_recorder new working bank: me: {} next_slot: {} next_leader: {}",
-                        my_pubkey,
-                        tpu_bank.slot(),
-                        next_leader
-                    );
-                    poh_recorder.lock().unwrap().set_bank(&tpu_bank);
-                }
+            // I guess I missed my slot
```

Comment:
Does this deserve a log or datapoint?

Reply:
there's a log above, could be changed to info!

I think this is somewhat rare (used to happen in local_cluster tests often, though) and harmless.